src/main.c

```diff
@@ -1132,28 +1136,38 @@ void handleMod3Key(KBDLLHOOKSTRUCT keyInfo, bool isKeyUp) {
 	if (isKeyUp) {
 		if (keyInfo.scanCode == scanCodeMod3R) {
 			level3modRightPressed = false;
+			double level3modRightPressedDuration = (double) (clock() - level3modRightPressedInstant) / CLOCKS_PER_SEC;
 			modState.mod3 = level3modLeftPressed | level3modRightPressed;
-			if (mod3RAsReturn && level3modRightAndNoOtherKeyPressed) {
+			if (mod3RAsReturn && level3modRightAndNoOtherKeyPressed && level3modRightPressedDuration < 0.5) {
 				sendUp(keyInfo.vkCode, keyInfo.scanCode, false); // release Mod3_R
 				sendDownUp(VK_RETURN, 28, true); // send Return
 				level3modRightAndNoOtherKeyPressed = false;
 			}
 		} else { // scanCodeMod3L (CapsLock)
 			level3modLeftPressed = false;
+			double level3modLeftPressedDuration = (double) (clock() - level3modLeftPressedInstant) / CLOCKS_PER_SEC;
 			modState.mod3 = level3modLeftPressed | level3modRightPressed;
-			if (capsLockAsEscape && level3modLeftAndNoOtherKeyPressed) {
+			if (capsLockAsEscape && level3modLeftAndNoOtherKeyPressed && level3modLeftPressedDuration < 0.5) {
 				sendUp(VK_CAPITAL, 58, false); // release Mod3_R
```

Comment:
I think this key (Mod3_R) should always be released, no matter if Escape is being sent or skipped.

Reply:
It should be the same in [line 1142](https://github.com/MaxGyver83/neo2-llkh/pull/42/files#diff-e0cf5b28d9b6b600f0af2bc78e8fd30ec675fd731a5da86f0c4283ffc0e40176R1142) above then, right? Since it's pretty much the same situation.

Reply:
Should I move the `sendUp`s out of the ifs in all three cases?